src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+            printInvalidMessage(type);
+        }
+    }
+
+    public static void showList() {
+        if (tasks.size() > 0) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < tasks.size(); i++) {
+                Task currentTask = tasks.get(i);
+                System.out.print(i+1 + ". " + currentTask.getTaskType() + currentTask.getStatusIcon() + " "
+                    + currentTask.getDescription());
+                if (currentTask.getTaskType() == "[D]")
+                    System.out.print(" (by: " + currentTask.getDateTime() + ")");
+                else if (currentTask.getTaskType() == "[E]") {
+                    System.out.print(" (at: " + currentTask.getDateTime() + ")");
+                }
+                System.out.print("\n");
+            }
+        } else {
+            System.out.println("Please input a task.");
+        }
+    }
+
+    public static void markAsDone(String taskDone) {
+        int taskIndex = Integer.parseInt(taskDone) - 1;
+        if (taskIndex >= tasks.size() || taskIndex < 0){
+            System.out.println("You have not added task " + taskIndex + " yet! Please try again.");
+        } else {
+            tasks.get(taskIndex - 1).setDone();
+            System.out.println("Nice! I've marked this task as done:");
+            Task currentTask = tasks.get(taskIndex);
+            System.out.println(currentTask.getTaskType() + currentTask.getStatusIcon()
+                    + currentTask.getDescription());
+        }
+    }
+
+    public static void addToDo(String taskToAdd) {
+        try {
+            validateInput(taskToAdd, "todo");
+            Todo t = new Todo(taskToAdd);
+            tasks.add(t);
+            System.out.println("Got it. I've added this task:");
+            System.out.println(t.getTaskType() + t.getStatusIcon() + " " + t.getDescription());
+            if (tasks.size() > 1) {
+                System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            } else {
+                System.out.println("Now you have " +tasks.size() + " task in the list.");
+            }
+        } catch (DukeException | IndexOutOfBoundsException e) {
+            printInvalidMessage("todo");
+        }
+    }
+
+    public static void addEvent(String taskToAdd) {
+        try {
+            validateInput(taskToAdd, "event");
+            int splitIndex = taskToAdd.indexOf('/');
+            String description = taskToAdd.substring(0, splitIndex);
+            String dateTime = taskToAdd.substring(splitIndex + 3);
+            Event e = new Event(description, dateTime);
+            tasks.add(e);
+            System.out.println("Got it. I've added this task:");
+            System.out.println(e.getTaskType() + e.getStatusIcon() + " " + e.getDescription() + " at(: "
+                    + e.getDateTime() + ")");
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
```

Comment:
Consider practicing SLAP and abstract this out into a method. Similarly you can do this for a lot of the printing response parts of the code.